src/BuildScriptGenerator/DotNetCore/ProjectFileProviders/ProbeAndFindProjectFileProvider.cs

```diff
@@ -63,26 +69,82 @@ public string GetRelativePathToProjectFile(RepositoryContext context)
 
             var webAppProjects = new List<string>();
             var azureFunctionsProjects = new List<string>();
+            var azureBlazorWasmProjects = new List<string>();
             var allProjects = new List<string>();
+            bool functionProjectExists = false;
+            bool webAppProjectExists = false;
+            bool blazorWasmProjectExists = false;
+
             foreach (var file in projectFiles)
             {
                 allProjects.Add(file);
-                if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                if (ProjectFileHelpers.IsAzureBlazorWebAssemblyProject(sourceRepo, file))
                 {
-                    webAppProjects.Add(file);
+                    azureBlazorWasmProjects.Add(file);
+                    blazorWasmProjectExists = true;
                 }
                 else if (ProjectFileHelpers.IsAzureFunctionsProject(sourceRepo, file))
                 {
                     azureFunctionsProjects.Add(file);
+                    functionProjectExists = true;
+                }
+                else if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                {
+                    webAppProjects.Add(file);
+                    webAppProjectExists = true;
                 }
             }
 
-            projectFile = GetProject(webAppProjects);
-            if (projectFile == null)
+            // Assumption: some build option "--apptype" will be passed to oryx
+            // which will indicate if the app is an azure function type or static type app
+
+            // If there are multiple projects, we will look for --appty to detect corresponding project.
+            // for example azurefunction and blazor both projects can reside
+            // at the same repo, so more than 2 csprojs will be found. Now we will
+            // look for --apptype value to determine which project needs to be built
+            string oryxAppTypeEnv = string.Empty;
+
+            if (_options != null
+                && _options.Value != null
+                && _options.Value.OryxAppType != null)
             {
-                projectFile = GetProject(azureFunctionsProjects);
+                oryxAppTypeEnv = _options.Value.OryxAppType;
+                _logger.LogInformation($"ProbeAndProjectFileProvider: {Constants.OryxAppType} is set to {oryxAppTypeEnv}");
+
+                if (functionProjectExists && oryxAppTypeEnv.ToLower().Contains("functions"))
+                {
+                    projectFile = GetProject(azureFunctionsProjects);
+                }
+                else if (blazorWasmProjectExists
+                    && (oryxAppTypeEnv.ToLower().Contains("static-sites")
+                    || oryxAppTypeEnv.ToLower().Contains("blazor-wasm")))
```

Comment:
I thought we should need only 2 app types for now: 'webapp' and 'static-sites'...webapp is for AppService's scenarios and 'static-sites' is for static webapps.

When 'static-sties' is used, the idea was to identity a maximum of 2 projects...one functions and one blazor-wasm and once if we find them we return those files and these files can then be used to publish them to different folders.

Let us chat about this today. We might want to handle this in a separate PR. I want us to at least get the part which Mitch is looking for now.